backend/xential/src/main/kotlin/com/ritense/valtimoplugins/xential/service/HttpClientConfig.kt

```diff
@@ -35,33 +33,42 @@ import javax.net.ssl.X509TrustManager
 
 class HttpClientConfig {
 
-    private fun trustManagerFactory(certFile: File): TrustManagerFactory {
+    private fun base64ToInputStream(base64String: String): ByteArrayInputStream {
+        // Decode the base64 encoded string to bytes
+        val decodedBytes = Base64.getDecoder().decode(base64String)
+
+        // Convert the decoded bytes array to ByteArrayInputStream
+        return ByteArrayInputStream(decodedBytes)
+    }
+
+    private fun trustManagerFactory(serverCertificate: String?): TrustManagerFactory? {
 
         // Load the server certificate
-        val certificateFactory = CertificateFactory.getInstance("X.509")
-        logger.info { "trustManagerFactory: Certificate file: $certFile" }
-        val serverCert = certificateFactory.generateCertificate(FileInputStream(certFile))
-
-        // Create a KeyStore with the server certificate
-        val trustStore = KeyStore.getInstance(KeyStore.getDefaultType()).apply {
-            load(null, null)
-            setCertificateEntry("server", serverCert)
-        }
+        if (serverCertificate != null) {
+            val certificateFactory = CertificateFactory.getInstance("X.509")
+            val clientCertificateDecoded = certificateFactory.generateCertificate(base64ToInputStream(serverCertificate))
+
+            // Create a KeyStore with the server certificate
+            val trustStore = KeyStore.getInstance(KeyStore.getDefaultType()).apply {
+                load(null, null)
+                setCertificateEntry("server", clientCertificateDecoded)
+            }
 
-        // Configure the TrustManager
-        val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm())
-        trustManagerFactory.init(trustStore)
-        return trustManagerFactory
+            // Configure the TrustManager
+            val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm())
+            trustManagerFactory.init(trustStore)
+            return trustManagerFactory
+        }
+        return null
     }
 
-    private fun keyManagerFactory(privateKeyFile: File?, clientCertFile: File?): KeyManagerFactory? {
-        return if (privateKeyFile != null && clientCertFile != null) {
+    private fun keyManagerFactory(clientPrivateKey: String?, clientCertificate: String?): KeyManagerFactory? {
+        return if (clientPrivateKey != null && clientCertificate != null) {
             val certificateFactory = CertificateFactory.getInstance("X.509")
```

Comment:
you could use kotlin's scope functions more as then there is no need to assign to local variables as those seem to be only used once.

something like

```suggestion
            CertificateFactory.getInstance("X.509").let { certificateFactory ->
    certificateFactory.generateCertificate(base64ToInputStream(clientCertificate)).let { clientCert ->
        loadPrivateKeyFromString(clientPrivateKey).let { privateKey ->
            // Create a KeyStore with the client certificate and private key
            KeyStore.getInstance(KeyStore.getDefaultType()).apply {
                load(null, null)
                setKeyEntry("client", privateKey, null, arrayOf(clientCert))
            }.let { keyStore -> 
                KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm()).apply {
                    init(keyStore, null)
                }
            }
        }
    }
}
```